Account/FxAClient10.swift

```diff
@@ -659,32 +651,35 @@ open class FxAClient10 {
 
     fileprivate func makeRequest<T>(_ request: URLRequest, responseHandler: @escaping (JSON) -> T?) -> Deferred<Maybe<T>> {
         let deferred = Deferred<Maybe<T>>()
+        urlSession.dataTask(with: request) { (data, response, error) in
+            guard let response = validatedHTTPResponse(response, contentType: "application/json") else {
+                deferred.fill(Maybe(failure: FxAClientError.local(FxAClientUnknownError)))
+                return
+            }
 
-        alamofire.request(request)
-            .validate(contentType: ["application/json"])
-            .responseJSON { response in
-                withExtendedLifetime(self.alamofire) {
-                    if let error = response.result.error {
-                        deferred.fill(Maybe(failure: FxAClientError.local(error as NSError)))
-                        return
-                    }
-
-                    if let data = response.result.value {
-                        let json = JSON(data)
-                        if let remoteError = FxAClient10.remoteError(fromJSON: json, statusCode: response.response!.statusCode) {
-                            deferred.fill(Maybe(failure: FxAClientError.remote(remoteError)))
-                            return
-                        }
-
-                        if let response = responseHandler(json) {
-                            deferred.fill(Maybe(success: response))
-                            return
-                        }
-                    }
-
-                    deferred.fill(Maybe(failure: FxAClientError.local(FxAClientUnknownError)))
-                }
-        }
+            if let error = error {
+                deferred.fill(Maybe(failure: FxAClientError.local(error as NSError)))
+                return
+            }
+
+            guard let data = data, !data.isEmpty else {
```

Comment:
@eoger if the data is empty, it throws `FxAClientUnknownError`, previously it threw a `remoteError`, maybe that is a bug?